src/data/p5.TypedDict.js

```diff
@@ -0,0 +1,371 @@
+/**
+ * @module Data
+ * @submodule Dictionary
+ * @for p5
+ * @requires core
+ */
+
+'use strict';
+
+var p5 = require('../core/core');
+
+/**
+ * This module defines the p5 methods for the p5 Dictionary classes
+ * these classes StringDict and NumberDict are for storing and working
+ * with key, value pairs
+ */
+
+/**
+ * Creates a new p5.StringDict using the key-value pair that you pass in
+ *
+ *
+ *
+ * @alt
+ * ALT TEXT HERE EVENTUALLY
+ */
+p5.prototype.createStringDict = function() {
+  return new p5.StringDict(arguments[0], arguments[1]);
+};
+
+// /**
+//  * Creates a new p5.NumberDict using the key-value pair that you pass in
+//  *
+//  * @method createNumberDict()
+//  * @param {}
+//  * @param {}
+//  * @return {p5.NumberDict}
+//  * @example
+//  * <div>
+//  * <code>
+//  * </code>
+//  * </div>
+//  *
+//  * @alt
+//  * ALT TEXT HERE EVENTUALLY
+//  */
+p5.prototype.createNumberDict = function() {
+  return new p5.NumberDict(arguments[0], arguments[1]);
+};
+
+p5.TypedDict = function() {
+};
+
+/**
+ * Returns the number of key-value pairs currently in Dictionary object
+ *
+ *
+ */
+p5.TypedDict.prototype.size = function(){
+  return this.count;
+};
+
+/**
+ * Returns true if dictionary has key value pair
+ *
+ *
+ */
+p5.TypedDict.prototype.hasKey = function(key) {
+  return this.data.hasOwnProperty(key);
+};
+
+/**
+ * Returns the key-value pair located at key
+ *
+ *
+ */
+p5.TypedDict.prototype.get = function(key) {
+  if(this.data.hasOwnProperty(key)){
+    return this.data[key];
+  }else{
+    console.log(key + ' does not exist in this Dictionary');
+  }
+};
+
+/**
+ * Changes the value of key if in Dictionary otherwise makes new one
+ *
+ *
+ */
+p5.TypedDict.prototype.set = function(key, value) {
+  if (arguments.length === 2) {
+    if(!this.data.hasOwnProperty(key)){
+      this.create(key, value);
+    } else {
+      this.data[key] = value;
+    }
+  }
+};
+
+
+p5.TypedDict.prototype._addObj = function(obj) {
+  for (var key in obj) {
+    if(this._validate(obj[key])) {
+      this.data[key] = obj[key];
+      this.count++;
+    }
+  }
+};
+
+
+/**
+ * Creates a new key-value pair in Dictionary object
+ *
+ *
+ */
+p5.TypedDict.prototype.create = function() {
+  if(arguments.length === 1) {
+    if(arguments[0] instanceof Object) {
+      this._addObj(arguments[0]);
+    }
+  }
+  else if(arguments.length === 2) {
+    var key = arguments[0];
```

Comment:
this code below looks a little familiar... could we just call this._addObj() here with the arguments obj?